sympy/solvers/decompogen.py

```diff
@@ -2,7 +2,7 @@
 from sympy.polys import Poly, decompose
 
 
-def decompogen(f, symbol):
+def decompogen(*args):
```

Comment:
Actually I was trying for [XFAIL](https://github.com/sympy/sympy/blob/master/sympy/solvers/tests/test_decompogen.py#L23) testcase, but I didn't find a way  for these types of cases.
I found that if we pass directly `lambda` function eg. `lambda x: x**2 + 2*x + 3` in current master branch then `function` variable contains `x**2 + 2*x + 3` so we can't extract lambda expression separately.
If we use `*args` then we can extract the `args` => `x` and `expr`=> `x**2 + 2*x + 3` using this PR changes.